Document enhanced tracing in Studio via OTel (#5967)
apollographql/router: docs/source/configuration/telemetry/apollo-telemetry.mdx

- Additional spans that were previously not included in Studio traces, such as query parsing, planning, execution, and more.
- Additional attributes including HTTP request details, REST connector details, and more.

It is expected that this will become the default in a future version of Router.

#### Configuration

This change adds a new configuration option `telemetry.apollo.experimental_otlp_tracing_sampler`. Use this option to send
a percentage of traces to Studio via OTLP instead of the native Apollo Usage Reporting protocol. Supported values:

- `always_off` (default): send all traces via the legacy Apollo Usage Reporting protocol.
- `always_on`: send all traces via OTLP.
- `0.0 - 1.0` (used for testing): the ratio of traces to send via OTLP (0.5 = 50 / 50).

Note that this sampler is only applied _after_ the common tracing sampler, for example:

#### Sample 1% of traces, send all traces via OTLP:

```yaml
telemetry:
  apollo:
    # Send all traces via OTLP
    experimental_otlp_tracing_sampler: always_on

  exporters:
    tracing:
      common:
        # Sample traces at 1% of all traffic
        sampler: 0.01
```

> OTel traces sent to Studio will not necessarily be identical to the ones sent to 3rd Party APM tools via OTLP:
> 
> - Only specific OTLP attributes will be included for parity with what is provided in legacy traces today. This ensures that data privacy
>   is maintained in an equivalent manner. The existing Router configuration options for Apollo telemetry will continue to function
>   with OTLP traces, such as forwarding of GraphQL errors, headers, and variables.
> - Some features of OTLP traces may only be available in Studio and not in 3rd Party APM tools (e.g. resolver-level timing information from
>   [Federated Tracing](../../federation/metrics/#enabling-federated-tracing)).

Reviewer: ```suggestion
OTLP traces sent to Studio aren't necessarily identical to ones sent to third-party APM tools via OTLP:

- Only specific OTLP attributes are included for parity with legacy traces today. This ensures that data privacy is maintained in an equivalent manner. Existing router configuration options for Apollo telemetry will continue to function with OTLP traces, including forwarding of GraphQL errors, headers, and variables.
- Some features of OTLP traces are available only in Studio and not in third-party APM tools, such as resolver-level timing information from [Federated Tracing](../../federation/metrics/#enabling-federated-tracing)
```

<Note>

> This change results in using a new wire protocol for traces, and some users may experience an increase in tracing traffic
> to GraphOS Studio due to the additional detail being captured. In exceptional situations it may be necessary to send fewer traces.
> This can be achieved via sending fewer traces (`telemetry.exporters.tracing.common.sampler`) or as a last resort, falling back
> to the old protocol via `telemetry.apollo.otlp_tracing_sampler` to send fewer OTLP traces or fully disable them.  
> Any performance regressions due to the new tracing protocol should also be reported to the Apollo support team.

Reviewer: ```suggestion
You may experience an increase in tracing traffic sent to GraphOS Studio due to the additional detail captured by the new wire protocol. In exceptional situations, you may need to send fewer traces.

To send fewer traces, configure `telemetry.exporters.tracing.common.sampler` or revert to the old protocol via `telemetry.apollo.otlp_tracing_sampler` to send fewer OTLP traces or to disable them.
  
For performance regressions due to the new tracing protocol, you should report them to the [Apollo support team](https://www.apollographql.com/support).
```

Support for OTel traces has historically only been available for 3rd party APM tools. With this option,
Studio can now provide a much more granular view of Router internals than the legacy Apollo tracing protocol.

Benefits of OTLP traces include:

- Comprehensive visualization of the router execution path in Studio
- New spans in Studio traces, including query parsing, planning, execution, and more
- New attributes, including HTTP request details, REST connector details, and more

#### Configuration

You can configure OTel traces with the  `telemetry.apollo.experimental_otlp_tracing_sampler` option. Use this option to send
a percentage of traces to Studio via OTLP instead of the native Apollo Usage Reporting protocol. Supported values:

- `always_off` (default): send all traces via the legacy Apollo Usage Reporting protocol
- `always_on`: send all traces via OTLP
- `0.0 - 1.0` (used for testing): the ratio of traces to send via OTLP (0.4 = 40% OTLP / 60% legacy)

This sampler is applied after the common tracing sampler.

#### Example configuration 

An example configuration that samples 1% of traces and sends all traces via OTLP:

> ```yaml

Reviewer 2: ```suggestion
```yaml title="router.config.yaml"
```

To make sure we're validating this configuration in our tests.
